### PR TITLE
[MRG] Fix Docker build for CI

### PR DESCRIPTION
**Pull request by standage**
We've had several cases over the last few months where the CI build (the Docker build in particular) failed for a contribution from the community or colleagues/collaborators without write access to the main repo. This PR fixes a bug in the Makefile and the Dockerfile and reads the full Github repository slug from `$TRAVIS_PULL_REQUEST_SLUG`.

Submitting this pull request from `standage/khmer` because obviously.

----------------------------------------

> <u>Message from the khmer maintainers</u>:
>
> *Please provide a description of the pull request above, including a
> reasonable level of detail and references to any relevant threads. After
> creating the  pull request, complete and mark the checklist below. If any item
> is not applicable, feel free to remove it or mark it as complete. If you are
> unsure about any item, feel free to ask about it before or during code
> review.*
>
> *When you are ready for the pull request to be reviewed, please post a comment
> with a message such as "Ready for review!"*

- [ ] Is any new functionality in tested? (This can be checked with
      `make clean diff-cover` or the CodeCov report that is automatically
      generated following a successful CI build.)
- [ ] <strike>Was a spellchecker run on the source code and documentation after
      changes were made?</strike>
- [ ] <strike>Have any changes to the command-line interface been explicitly described?
      Only backwards-compatible additions are allowed without a major version
      increment. Changing file formats also requires a major version number
      increment.</strike>
- [ ] <strike>Have any substantial changes been documented in `CHANGELOG.md`? See
      [keepachangelog](http://keepachangelog.com/) for more details.</strike>
- [ ] <strike>Do the changes respect streaming I/O? (Are they tested for streaming I/O?)</strike>

### CODE REVIEW

**Comment by standage** [comment]
This is the reason @ctSkennerton's second commit in #1863 failed.

**Comment by standage** [comment]
Ready for review and merge! @ctb @camillescott @luizirber @betatim

File: docker/Dockerfile [hunk 1]
```diff
@@ -12,5 +12,7 @@ WORKDIR /home
 
 ARG branch
 ENV branch ${branch:-v2.1.2}
+ARG slug
+ENV slug ${slug:-dib-lab/khmer}
```
Comment (betatim) [inline docker/Dockerfile]: This is the bit that makes sure to set `slug` to dib-lab/khmer when `TRAVIS_PULL_REQUEST_SLUG` is empty?

**Comment by betatim** [comment]
I did not know `TRAVIS_PULL_REQUEST_SLUG` was a thing!

File: docker/Dockerfile (hunk 1 above)
Comment (standage) [reply docker/Dockerfile]: Correct!

**Comment by standage** [comment]
Thanks.

**Comment by ctb** [comment]
Nice work!

**Pull request merged**